### ISSUE TITLE
Choose perl interpreter for rule?

### ISSUE THREAD

**Issue by pawel125** [snakemake/snakemake#547]
Hi!

I am running ensembl-vep and when I run it outside the snakemake it works correctly, but as a part of the workflow it fails due to lacking perl packages. I suppose that inside the snakemake it uses conda's (I have used conda to install snakemake) perl interpreter, which do not have these packages installed. Unfortunatelly, we have a problem with installing perl packages on our cluster due to file locking problems in Lustre. Therefore I would like vep to use the default perl interpreter which has these packages in some way installed.

How can I ensure that the default perl interpreter is used inside the certain rules?

**Comment by elsherbini** [comment]
I'm not super familiar with perl, but the first thing i'd try is aliasing perl to be your default perl inside the rule

```
rule vep:
input: ...
output: ...
shell:
"""
alias perl=/path/to/perl;
vep ...
"""
If conda is messing up your PERL5LIB path which is preventing vep from getting what it needs, you might need to add a `export PERL5LIB=/path/to/perl/libs` to the shell directive.

**Comment by pawel125** [comment]
Thank you for your answer!

Aliasing did not change anything, still conda's perl was in use. Fortunatelly I have found some workaround (below). Also, when I tried to echo PERL5LIB variable in my rule I get 
> /usr/bin/bash: line 2: PERL5LIB: unbound variable

But PERL5LIB does not mean which perl to use, but where should it look for the packages, am I right? If so, can I tell conda's perl to use packages installed in default system perl?

**Solution**

I solved the problem by adding /usr/bin to the begining of the path `export PATH=/usr/bin:$PATH`. Looks like a workaround, I thought that I should point the rigth version in environment.yml file, but I could not get it woriking this way.
